wasmcloud-wasi-kv/wasmcloud-wasi-kv-wit/src/lib.rs

```diff
@@ -0,0 +1,420 @@
+wit_bindgen::generate!({
+    path: "../wit",
+    world: "keyvalue",
+});
+
+use std::borrow::{Borrow, BorrowMut};
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::thread::LocalKey;
+use std::hash::Hash;
+use std::sync::RwLock;
+use std::str::from_utf8;
+
+use crate::types::{IncomingValue, InputStream, OutgoingValue, OutputStream};
+use crate::keyvalue::{Bucket, Error};
+use crate::wasmcloud_guest::WasmcloudGuest;
+use crate::wasmcloud_host::{
+    host_call as wasmcloud_host_call,
+};
+
+use wasmcloud_interface_keyvalue::{
+    GetResponse,
+    SetRequest,
+    encode_set_request,
+};
+
+thread_local! {
+    pub static OUTGOING_VALUE: RefCell<OutgoingValue> = RefCell::new(0 as OutgoingValue);
+    pub static OUTGOING_VALUES: RwLock<HashMap<OutgoingValue, Vec<u8>>> = RwLock::new(HashMap::new());
+
+    pub static INCOMING_VALUE: RefCell<IncomingValue> = RefCell::new(0);
+    pub static INCOMING_VALUES: RwLock<HashMap<IncomingValue, Vec<u8>>> = RwLock::new(HashMap::new());
+
+    pub static INPUT_STREAM: RefCell<InputStream> = RefCell::new(0);
+    pub static INPUT_STREAMS: RwLock<HashMap<InputStream, IncomingValue>> = RwLock::new(HashMap::new());
+
+    pub static OUTPUT_STREAM: RefCell<OutputStream> = RefCell::new(0);
+    pub static OUTPUT_STREAMS: RwLock<HashMap<OutputStream, OutgoingValue>> = RwLock::new(HashMap::new());
+
+    pub static BUCKET: RefCell<Bucket> = RefCell::new(0);
+    pub static BUCKETS: RwLock<HashMap<Bucket, String>> = RwLock::new(HashMap::new());
+}
```

Comment:
Could you comment a little here on the use of input/output streams to handle these keyvalue operations? It's a bit of a different pattern than I'm used to (just directly handling the operation)

Reply:
Yeah sure can!

- [ ] Add comments explaining the WIT input/output stream paradigms